### PR TITLE
add attendee pr review

### PR DESCRIPTION
**Pull request by humbertouchiyama**

### CODE REVIEW

File: app/src/main/java/com/humberto/tasky/agenda/data/event/EventMappers.kt [hunk 1]
```diff
@@ -34,13 +42,9 @@ fun AgendaItem.Event.toEventEntity(): EventEntity {
...
-        attendees = attendees.map {
-            it.toLocalAttendee(
-                eventId = id,
-                remindAt = remindAtEpoch
-            )
-        },
-        photoKeys = photos.map { it.key }
+        attendees = attendees.map { it.toLocalAttendee() },
+        photoKeys = photos.map { it.key },
+        host = eventCreator?.userId ?: hostId ?: ""
```
Comment (humbertouchiyama) [inline app/src/main/java/com/humberto/tasky/agenda/data/event/EventMappers.kt:47]: not sure about this 😅

File: app/src/main/java/com/humberto/tasky/agenda/data/event/EventMappers.kt [hunk 2]
```diff
@@ -9,7 +9,9 @@ import com.humberto.tasky.core.database.entity.PhotoEntity
 import com.humberto.tasky.core.domain.util.toZonedDateTime
 
 fun EventEntity.toEvent(
-    photos: List<Photo>
+    photos: List<Photo>,
+    eventCreator: LocalAttendee? = null,
+    localAttendee: LocalAttendee? = null
 ): AgendaItem {
```
Comment (Kai-Cilliers) [inline app/src/main/java/com/humberto/tasky/agenda/data/event/EventMappers.kt:15]: I wouldn’t accept nullable type here, an event has to have an event creator

File: app/src/main/java/com/humberto/tasky/agenda/data/event/EventMappers.kt [hunk 3]
```diff
@@ -18,13 +20,19 @@ fun EventEntity.toEvent(
...
         photos = photos,
-        isUserEventCreator = isUserEventCreator
+        isUserEventCreator = isUserEventCreator,
+        eventCreator = eventCreator?.toAttendee(isEventCreator = true),
+        localAttendee = localAttendee?.toAttendee(isEventCreator = localAttendee.userId == eventCreator?.userId)
     )
 }
 
-fun AgendaItem.Event.toEventEntity(): EventEntity {
+fun AgendaItem.Event.toEventEntity(hostId: String?): EventEntity {
```
Comment (Kai-Cilliers) [inline app/src/main/java/com/humberto/tasky/agenda/data/event/EventMappers.kt:35]: Same here, is there a reason why you have this as nullable?

File: app/src/main/java/com/humberto/tasky/agenda/data/event/EventMappers.kt (hunk 1 above)
Comment (Kai-Cilliers) [reply app/src/main/java/com/humberto/tasky/agenda/data/event/EventMappers.kt:47]: Don’t accept a nullable type 😄 See my previous comment

File: app/src/main/java/com/humberto/tasky/agenda/data/event/EventRepositoryImpl.kt [hunk 4]
```diff
@@ -6,36 +6,65 @@ import com.humberto.tasky.agenda.domain.event.Attendee
...
                 .getPhotosByKeys(eventEntity.photoKeys)
                 .map { it.toPhoto() }
+            val localAttendee = event.attendees.find { it.userId == localUserId }
+            val eventCreator = event.attendees.find { it.userId == eventEntity.host }
             Result.Success(
                 eventEntity.toEvent(
-                    photos = photos
+                    photos = photos,
+                    eventCreator = eventCreator,
+                    localAttendee = localAttendee
```
Comment (Kai-Cilliers) [inline app/src/main/java/com/humberto/tasky/agenda/data/event/EventRepositoryImpl.kt:36]: I see your `EventEntity` has attendees, why then do you pass in the attendees to `toEvent`?
Regardless, if the `eventCreator` is null, then you can consider the data invalid and return an error - the event should always have a creator

**Pull request merged**